### ISSUE TITLE
WPcom Editing Toolkit causes fatal errors on self-hosted sites

### ISSUE THREAD

**Issue by cuemarie** [Automattic/wp-calypso#77573]
### Quick summary

If I activate WordPress.com Editing Toolkit on my self-hosted site (hosted with Pressable), I encounter several critical PHP errors. 

First, activating the plugin with no other plugins active results in the following error:

```
PHP Fatal error: Uncaught Error: Class "Jetpack_Options" not found in /srv/htdocs/wp-content/plugins/full-site-editing/help-center/class-help-center.php:139
```

After deactivating, I activated and connected Jetpack on my site, then tried again. This time, I encountered this error:

```
PHP Fatal error: Uncaught Error: Call to undefined function A8C\FSE\wpcom_get_site_purchases() in /srv/htdocs/wp-content/plugins/full-site-editing/help-center/class-help-center.php:143
```

These errors seem to imply that this plugin cannot work on a self-hosted site, but we don't make that known anywhere on the plugin's page.  The reviews for the plugin also [reflect this poor experience](https://wordpress.org/support/plugin/full-site-editing/reviews/)

### Steps to reproduce

1. On a self-hosted site, install and activate the [WordPress.com Editing Toolkit plugin](https://wordpress.org/plugins/full-site-editing/)

### What you expected to happen

Either the plugin would work, or it would warn me that it cannot work outside of WordPress.com

### What actually happened

A series of fatal errors, noted above.

### Impact

Some (< 50%)

### Available workarounds?

No and the platform is unusable

### Platform (Simple and/or Atomic)

Self-hosted

### Logs or notes

- WPcom Editing Toolkit v3.67126

**Comment by cuemarie** [comment]
Fixed by https://github.com/Automattic/wp-calypso/pull/81967

**Comment by cuemarie** [comment]
I tested this to close yesterday, but returned to my self-hosted site today to find it fatal-ing after all.  Currently have the following plugins active:
- Gutenberg 16.9.0
- Jetpack 12.8-a.7
- WordPress.com Editing Toolkit 4.2070

⚠️ **ERROR**

```
PHP Fatal error: Uncaught Error: Class "Domain_Mapping" not found in /srv/htdocs/wp-content/plugins/full-site-editing/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php:153 Stack trace: #0 /srv/htdocs/wp-content/plugins/full-site-editing/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php(87): A8C\FSE\WPCOM_Domain_Upsell_Callout->blog_has_custom_domain(194267179) #1 /srv/htdocs/wp-content/plugins/full-site-editing/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php(45): A8C\FSE\WPCOM_Domain_Upsell_Callout->should_not_show_callout() #2 /wordpress/core/6.3.2/wp-includes/class-wp-hook.php(310): A8C\FSE\WPCOM_Domain_Upsell_Callout->enqueue_script_and_style('') #3 /wordpress/core/6.3.2/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters(NULL, Array) #4 /wordpress/core/6.3.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #5 /wordpress/core/6.3.2/wp-admin/edit-form-blocks.php(272): do_action('enqueue_block_e...') #6 /wordpress/core/6.3.2/wp-admin/post-new.php(72): require('/wordpress/core...') #7 {main} thrown in /srv/htdocs/wp-content/plugins/full-site-editing/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php on line 153
```

**Comment by obenland** [comment]
@lupus2k Could you take a look at this when you get a chance? Looks like it just needs a class_exists check before using it.

**Comment by katiebethbrown** [comment]
Wanted to note that this is still occurring. 

Deactivating Jetpack on an Atomic site while ETK is active causes a fatal in wp-admin:

```
[15-Mar-2024 02:25:02 UTC] PHP Fatal error:  Uncaught Error: Class "Jetpack_Options" not found in /wordpress/plugins/full-site-editing/4.15835/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php:84
Stack trace:
#0 /wordpress/plugins/full-site-editing/4.15835/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php(45): A8C\FSE\WPCOM_Domain_Upsell_Callout->should_not_show_callout()
#1 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): A8C\FSE\WPCOM_Domain_Upsell_Callout->enqueue_script_and_style('')
#2 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#3 /wordpress/core/6.4.3/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#4 /wordpress/core/6.4.3/wp-admin/edit-form-blocks.php(272): do_action('enqueue_block_e...')
#5 /wordpress/core/6.4.3/wp-admin/post-new.php(72): require('/wordpress/core...')
#6 {main}
  thrown in /wordpress/plugins/full-site-editing/4.15835/wpcom-domain-upsell-callout/class-wpcom-domain-upsell-callout.php on line 84
```

**Comment by obenland** [comment]
@lupus2k This module should probably not run on self-hosted sites. Could add a check that avoids the plugin fataling on self-hosted sites?

**Comment by katiebethbrown** [comment]
@obenland @lupus2k - If we're adding a conditional for self-hosted sites, can we extend that so things also fail gracefully on Atomic sites if Jetpack gets disabled?

**Comment by obenland** [comment]
Maybe we could [declare a dependency to Jetpack](https://make.wordpress.org/core/2024/03/05/introducing-plugin-dependencies-in-wordpress-6-5/), so the plugin has to be deactivated before deactivating Jetpack.

**Comment by taipeicoder** [comment]
The ETK plugin is now deprecated. @obenland could you advise on the next steps to delist the plugin from .org? cc: @arthur791004

**Comment by obenland** [comment]
Is it no longer a managed plugin on Atomic? pekYwv-4As-p2#comment-4238

**Comment by taipeicoder** [comment]
@arthur791004 did we fix that via D162416-code?

**Comment by arthur791004** [comment]
Yes. The ETK plugin won't be installed on the newly created atomic sites anymore.

**Comment by dsas** [comment]
And the existing atomic sites have been cleaned up too: pbxlJb-6ay-p2#comment-3905 So we can now delisted from .org

There are some instructions here on how to remove (close) a plugin: https://developer.wordpress.org/plugins/wordpress-org/plugin-developer-faq/#closed-plugins

**Comment by obenland** [comment]
There were still some 6k sites that had the plugin, last I checked. Please don't remove the plugin from the repo until we know for sure it's no longer managed on Atomic.

**Comment by taipeicoder** [comment]
@arthur791004 could you follow-up on this? Thanks!

**Comment by arthur791004** [comment]
Okay! BTW I've asked in p1729074261619949-slack-C7YPW6K40 about the issue that ~6k sites still have the managed plugin.

**Comment by taipeicoder** [comment]
We should be able to remove the ETK plugin from the remaining sites via pbfA90-1Ou-p2.

**Comment by arthur791004** [comment]
Asked p1736422890861299/1729074261.619949-slack-C7YPW6K40 again to see whether the Task API is available to delete ETK plugins on rest of atomic sites.

**Comment by arthur791004** [comment]
Executed the task, 169775-ghe-Automattic/wpcom, to clean up the ETK plugin on atomic sites again.

**Comment by taipeicoder** [comment]
A summary of the efforts we've done so far:
- On September 2024 (pbxlJb-6ay-p2#comment-3905), we removed the ETK plugin from all WoA sites.
- On October 2024, it was [pointed out](https://github.com/Automattic/wp-calypso/issues/77573#issuecomment-2442203210) that ~6k sites still had the plugin.
- On early January 2025, we ran another ETK clean-up. And followed up (p1737012152862919-slack-CDLH4C1UZ) with the Jetpack team to see if Jetpack Backup was re-installing the ETK plugin when restoring sites. The following issue was created https://github.com/Automattic/dotcom-forge/issues/10262.
- On late January 2025, the Atomic team noticed (p1737565339870709-slack-CRWCHQGUB) some errors caused by the latest ETK clean-up. We fixed this issue via the PRs https://github.com/Automattic/jetpack/pull/41273 and https://github.com/Automattic/jetpack/pull/41887.

In short, we still need to look into why ETK plugin is seemly being restored via Jetpack Backup before removing it from the .org repo.
